third-party/cmake/BuildLibuv.cmake

```diff
@@ -1,18 +1,77 @@
-ExternalProject_Add(libuv
-  PREFIX ${DEPS_BUILD_DIR}
-  URL ${LIBUV_URL}
-  DOWNLOAD_DIR ${DEPS_DOWNLOAD_DIR}/libuv
-  DOWNLOAD_COMMAND ${CMAKE_COMMAND}
-    -DPREFIX=${DEPS_BUILD_DIR}
-    -DDOWNLOAD_DIR=${DEPS_DOWNLOAD_DIR}/libuv
-    -DURL=${LIBUV_URL}
-    -DEXPECTED_SHA256=${LIBUV_SHA256}
-    -DTARGET=libuv
-    -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/DownloadAndExtractFile.cmake
-  CONFIGURE_COMMAND sh ${DEPS_BUILD_DIR}/src/libuv/autogen.sh &&
-    ${DEPS_BUILD_DIR}/src/libuv/configure --with-pic --disable-shared
-      --prefix=${DEPS_INSTALL_DIR} --libdir=${DEPS_INSTALL_DIR}/lib
-      CC=${DEPS_C_COMPILER}
-  INSTALL_COMMAND ${MAKE_PRG} install)
+# Reusable function to build libuv, the single argument is
+# the target name. The following variables are used as build
+# commands:
+#   LIBUV_CONFIGURE_COMMAND
+#   LIBUV_BUILD_COMMAND
+#   LIBUV_INSTALL_COMMAND
+#
+function(BuildLibuv TARGET_NAME)
+  ExternalProject_Add(${TARGET_NAME}
+    PREFIX ${DEPS_BUILD_DIR}
+    URL ${LIBUV_URL}
+    DOWNLOAD_DIR ${DEPS_DOWNLOAD_DIR}/libuv
+    DOWNLOAD_COMMAND ${CMAKE_COMMAND}
+      -DPREFIX=${DEPS_BUILD_DIR}
+      -DDOWNLOAD_DIR=${DEPS_DOWNLOAD_DIR}/libuv
+      -DURL=${LIBUV_URL}
+      -DEXPECTED_SHA256=${LIBUV_SHA256}
+      -DTARGET=${TARGET_NAME}
+      -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/DownloadAndExtractFile.cmake
+    CONFIGURE_COMMAND "${LIBUV_CONFIGURE_COMMAND}"
+    BUILD_COMMAND "${LIBUV_BUILD_COMMAND}"
+    INSTALL_COMMAND "${LIBUV_INSTALL_COMMAND}")
+endfunction()
+
+set(UNIX_CFGCMD sh ${DEPS_BUILD_DIR}/src/libuv/autogen.sh &&
+      ${DEPS_BUILD_DIR}/src/libuv/configure --with-pic --disable-shared
+        --prefix=${DEPS_INSTALL_DIR} --libdir=${DEPS_INSTALL_DIR}/lib
+        CC=${DEPS_C_COMPILER}
+        )
+
+if (UNIX)
+  set(LIBUV_CONFIGURE_COMMAND ${UNIX_CFGCMD})
+  set(LIBUV_BUILD_COMMAND "")
+  set(LIBUV_INSTALL_COMMAND ${MAKE_PRG} install)
+  BuildLibUv(libuv) 
+
+elseif (MINGW AND CMAKE_CROSSCOMPILING)
+  # Build libuv for the host
```

Comment:
I have not been able to do it yet, but that is the main idea. It will require something like wine to run the target - unfortunately I have tried this before a wine blocked the build job in travis.